### PR TITLE
Added option to use datadog tracer for middlewares

### PR DESCRIPTION
**Pull request by karlssonerik**

### CODE REVIEW

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -17,4 +17,5 @@ jobs:
       - name: Run linter
         uses: golangci/golangci-lint-action@v2
         with:
-          version: v1.45
+          version: v1.52
+          args: --timeout=2m
```
Comment (Ekberg) [inline .github/workflows/linting.yml:21]: Can we configure this in the yaml config instead?

File: trace.go [hunk 2]
```diff
@@ -36,8 +39,7 @@ func (s openCensusSpan) AddStringAttribute(name, value string) {
 	s.span.AddAttributes(trace.StringAttribute(name, value))
 }
 
-func (t *OpenCensusTracer) StartSpan(ctx context.Context, resourceName string) (context.Context, Span) {
-	// Avoid creating a new trace for the middlewares, most requests will have a trace but
+func (t *OpenCensusTracer) StartSpan(ctx context.Context, resourceName string) (context.Context, Span) { // Avoid creating a new trace for the middlewares, most requests will have a trace but
```
[the remark this answers is not in the document]
Comment (karlssonerik) [reply trace.go:42]: No

File: trace.go [hunk 3]
```diff
@@ -56,3 +58,54 @@ func (t *OpenCensusTracer) SpanFromContext(ctx context.Context) Span {
...
+func (t *DataDogTracer) StartSpan(ctx context.Context, resourceName string) (context.Context, Span) {
+	// health endpoints will not. This will avoid creating unnessesary traces for those.
+	_, isFound := ddtracer.SpanFromContext(ctx)
+	if !isFound {
+		return ctx, &NilSpan{}
+	}
+
+	ddOperationName, ddResourceName := parseResourceNameToDDNames(resourceName)
+
+	span, ctx := ddtracer.StartSpanFromContext(ctx, ddOperationName, ddtracer.ResourceName(ddResourceName))
```
[the remark this answers is not in the document]
Comment (karlssonerik) [reply trace.go:82]: Yeah I had, "web.middleware" as constant at first, I guess I can easily change it. But I think the big problem was there was not a obvious way to map 1:1 from the old trace path the dd names.

Now it Middleware/Authentication/isAuthenticationNeeded becomes "middleware.authentication: isAuthenticationNeeded"

maybe "web.middleware: Authentication.isAuthenticationNeeded" is better?

File: .github/workflows/linting.yml (hunk 1 above)
Comment (karlssonerik) [reply .github/workflows/linting.yml]: moved

File: trace.go [hunk 4]
```diff
@@ -56,3 +58,54 @@ func (t *OpenCensusTracer) SpanFromContext(ctx context.Context) Span {
 
 	return openCensusSpan{span: span}
 }
+
+type DataDogTracer struct{}
```
[the remark this answers is not in the document]
Comment (karlssonerik) [reply trace.go:63]: Do you mean by checking what span you van fins in the context?

Maybe in another PR?

**Comment by karlssonerik** [comment]
Closing this in favour of #27 and #28

**Pull request closed without merging**